tests/misc_test.go

```diff
@@ -635,3 +635,32 @@ func TestTypeConversion(t *testing.T) {
 		t.Fail()
 	}
 }
+
+// See https://github.com/gopherjs/gopherjs/issues/851.
+func TestSlicingNilSlice(t *testing.T) {
+	t.Run("StaysNil", func(t *testing.T) {
+		var s []int
+		s = s[:]
+		s = s[0:0]
+		s = s[0:0:0]
```

Comment:
I'd add the same `if s != nil` above this

Reply:
Are you suggesting this in case one of the assigments makes `s` non-nil, but the following line makes it nil again, and we don't catch it?

Reply:
As `s` is expected to be non-nil, you don't have to reassign nil, but probably reassiging would be better. Thanks!